### PR TITLE
Feat(event): Update Function For Object

### PR DESCRIPTION
**Pull request by johndy2742**
### Changes Made

- add update object logic

### Purpose

- add update functionality for Object managament

### How Has This Been Tested

- Postman request test
image

### Related Links

https://chat.digitalservice.id/boards/team/nii5pbyw1idwxrxjd8bp56mtkw/bte55h31wfjgp9xgry174ur3mae/vbyyag4aadi8ejgca6gtqzs55gc

Project: Event Management System
Participants: @johndy2742, @iqbal167, @alqinsidev, @MNAlfiansyah , @ristianaditya

### CODE REVIEW

File: service/src/modules/object/repository/object.go [hunk 1]
```diff
@@ -150,3 +150,27 @@ func (r *Repository) filterObjectQuery(params request.QueryParam, binds *[]inter
 
 	return query
 }
+
+func (r *Repository) UpdateObject(ctx context.Context, obj request.Object) (request.Object, error) {
```
Comment (iqbal167) [inline service/src/modules/object/repository/object.go:154]: please user pointer @johndy2742, and why still use request as a struct?

Comment (johndy2742) [reply service/src/modules/object/repository/object.go:154]: i still have the same problem as before where if i use struct without json tag like entity the insertion wont work kang

File: service/src/modules/object/repository/object.go [hunk 2]
```diff
@@ -150,3 +150,27 @@ func (r *Repository) filterObjectQuery(params request.QueryParam, binds *[]inter
...
+        WHERE "id" = $1
+    `
+
+	_, err = r.db.Master.ExecContext(ctx,
+		query,
+		obj.ID, obj.Name, obj.Address, obj.Description, bannerArray, obj.Logo, socialMediaJSON, obj.Organizer, obj.Status, time.Now(),
+	)
+
+	if err != nil {
+		return &request.Object{}, err
```
Comment (iqbal167) [inline service/src/modules/object/repository/object.go:172]: just set`return nil, err`

File: service/src/modules/object/usecase/update_object.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	"context"
+
+	"github.com/jabardigitalservice/super-app-services/event/src/modules/object/transport/handler/http/request"
+)
+
+func (uc *Usecase) UpdateObject(ctx context.Context, obj *request.Object) (*request.Object, error) {
+	updatedObj, err := uc.repo.UpdateObject(ctx, obj)
+
+	if err != nil {
+		return &request.Object{}, err
```
Comment (iqbal167) [inline service/src/modules/object/usecase/update_object.go:13]: just setreturn nil, err

File: service/src/modules/object/repository/object.go [hunk 4]
```diff
@@ -150,3 +150,27 @@ func (r *Repository) filterObjectQuery(params request.QueryParam, binds *[]inter
 
 	return query
 }
+
+func (r *Repository) UpdateObject(ctx context.Context, obj *request.Object) (*request.Object, error) {
+	socialMediaJSON, err := json.Marshal(obj.SocialMedia)
+	if err != nil {
+		return &request.Object{}, err
```
Comment (iqbal167) [inline service/src/modules/object/repository/object.go:157]: please return nil, err too

**Review by iqbal167 — APPROVED** [review]

**Pull request merged**